src/js/tests/filter-form-mock-html.js

```diff
@@ -0,0 +1,176 @@
+exports.PostsFilterFormTestHTML = `
+  <aside class="mzp-l-sidebar custom-width">
+
+    <details class="filter-list-sidebar-content filter-list-sidebar-content-mobile">
+      <summary>Filter</summary>
+
+      <form class="filter-form js-filter-form" data-controls="article_or_video-cards" action="/posts/">
+
+        <header class="filter-form-section js-filter-form-clear-section">
+          <a href="#" class="filter-form-clear js-filter-clear">
+            <span class="icon">
+              <svg width="12" height="12" xmlns="http://www.w3.org/2000/svg"><path d="M6 4.586L10.243.343a1 1 0 1 1 1.414 1.414L7.414 6l4.243 4.243a1 1 0 1 1-1.414 1.414L6 7.414l-4.243 4.243a1 1 0 1 1-1.414-1.414L4.586 6 .343 1.757A1 1 0 1 1 1.757.343L6 4.586z" fill="currentColor" fill-rule="evenodd"></path></svg>
+
+            </span>
+            Clear all
+          </a>
+        </header>
```

Comment:
I would probably just use a `button` element here and no use the header at all. A `header` element without any `H` elements seems a little non-semantic ;)